src/devices/PiJuice/PiJuicePower.cs

```diff
@@ -0,0 +1,121 @@
+{
+    /// <summary>
+    /// PiJuicePower class to support status of the PiJuice
+    /// </summary>
+    public class PiJuicePower
+    {
+        private readonly PiJuice _piJuice;
+
+        /// <summary>
+        /// PiJuicePower constructor
+        /// </summary>
+        /// <param name="piJuice">The PiJuice class</param>
+        public PiJuicePower(PiJuice piJuice)
+        {
+            _piJuice = piJuice;
+        }
+
+        /// <summary>
+        /// The delay before the PiJuice removes power to the GPIO pins
+        /// </summary>
+        /// <value>The delay in seconds</value>
+        public TimeSpan PowerOff
+        {
+            get
+            {
+                var response = _piJuice.ReadCommand(PiJuiceCommand.PowerOff, 1);
+
+                return new TimeSpan(0, 0, response[0]);
+            }
+            set
+            {
+                if (value.TotalSeconds < 0 || value.TotalSeconds > 255)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value.TotalSeconds));
+                }
+
+                _piJuice.WriteCommand(PiJuiceCommand.PowerOff, new byte[] { (byte)value.TotalSeconds });
+            }
+        }
+
+        /// <summary>
+        /// The current state of the Wakeup on charge
+        /// </summary>
+        /// <returns>Current state of the wake up on charge function</returns>
+        public WakeUpOnCharge WakeUpOnCharge
+        {
+            get
+            {
+                var response = _piJuice.ReadCommand(PiJuiceCommand.WakeUpOnCharge, 1);
+
+                return new WakeUpOnCharge
+                {
+                    Disabled = response[0] == 0x7F,
+                    WakeUpPercentage = new Ratio((response[0] == 0x7F ? 0 : response[0]), UnitsNet.Units.RatioUnit.Percent)
+                };
+            }
+            set
+            {
+                if (!value.Disabled && (value.WakeUpPercentage.Percent < 0 || value.WakeUpPercentage.Percent > 100))
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value.WakeUpPercentage));
+                }
+
+                _piJuice.WriteCommandVerify(PiJuiceCommand.WakeUpOnCharge, new byte[] { (byte)(value.Disabled ? 0x7F : (short)value.WakeUpPercentage.Percent) });
+            }
+        }
+
+        /// <summary>
+        /// The current watchdog timer time after which it will power cycle if it does not receive a heartbeat signal
+        /// </summary>
+        /// <value>Time in minutes after which PiJuice will power cycle if it does not receive a heartbeat signal </value>
+        public TimeSpan WatchdogTimer
+        {
+            get
+            {
+                var response = _piJuice.ReadCommand(PiJuiceCommand.WatchdogActiviation, 2);
+
+                return new TimeSpan(0, BinaryPrimitives.ReadInt16LittleEndian(response), 0);
+            }
+            set
+            {
+                if (value.TotalMinutes < 0 || value.TotalMinutes > 65535)
```

Comment:
upper limit seems a bit high, it's 45 days

Reply:
Its in the original python code, https://github.com/PiSupply/PiJuice/blob/d3c193c9f40ffc2f45cdb9304cd7c59ee46952b2/Software/Source/pijuice.py#L955